src/Vault.sol

```diff
@@ -759,6 +776,10 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, Ownable {
     if (_amountOut > _liquidatableYield)
       revert LiquidationAmountOutGTYield(_amountOut, _liquidatableYield);
 
+    uint256 _vaultMaxMint = _getVaultMaxMint(_totalSupply());
+    if (_amountOut > _vaultMaxMint)
+      revert LiquidationAmountOutGTVaultMaxMint(_amountOut, _vaultMaxMint);
```

Comment:
I believe this might be redundant- because the TwabController should catch the overflow.

_totalSupply() is using the twab controller total supply, which means this code is checking to see whether the twab total supply is less than uint112. This is already handled indirectly by the TwabController here: https://github.com/GenerationSoftware/pt-v5-twab-controller/blob/e81e29c8bb6f374ecce1c4f5beb0be146dc56f67/src/libraries/TwabLib.sol#L107

This call adds quite a bit of extra gas for a case that's already handled

Reply:
Yes but at least we have a comprehensive error message.
I can remove if we don't want it.